src/npc.h

```diff
@@ -1391,6 +1412,169 @@ class npc_template
         static void check_consistency();
 };
 
+class npc_work_log_entry
+{
+    private:
+        // required
+        std::string entry_npc_name;
+        time_point entry_timestamp;
+        int entry_npc_id;
+        log_entry_type entry_type;
+        std::string entry_description;
+        // not required
+        cata::optional<tripoint> entry_position;
+        cata::optional<activity_reason_info> entry_reason;
+    public:
+        npc_work_log_entry() {
+            entry_npc_name = "nullname";
+            entry_timestamp = calendar::before_time_starts;
+            entry_npc_id = -1;
+            entry_type = NUM_ENTRY_TYPE;
+            entry_description = "nulldesc";
+        }
+        npc_work_log_entry( npc &p, log_entry_type new_entry_type, std::string new_entry_description ) {
+            entry_timestamp = calendar::turn;
+            entry_npc_name = p.name;
```

Comment:
Can this value be retrieved on-spot using `entry_npc_id` instead?

Reply:
Sorry, not sure what you mean?

Reply:
I mean if you save the NPC id then there's no need to save the name, you can retrieve the name when you need it, using the NPC id?

Reply:
Ah yes, anything that needs the name can just get it from that, ill do that.

Reply:
I see a problem with this actually : retrieving the character from their ID presumes that the character is still alive, which they may not be, its best to store the name historically.

Reply:
Oh, I see, thanks for checking.